File: python/sparklabs/batch/sales/dataframe/distinct_customer.py
from pyspark.sql import functions as F

from sparklabs.batch.sales.dataframe import utils


def run() -> None:
    distinct_customer_count = utils.orders.select(F.col("customer_id")).count()
    expensive_product = utils.products.orderBy(F.desc("price")).limit(1)

    print(f"Total distinct customers: {distinct_customer_count}")
    print("Most Expensive Product:")
    expensive_product.show(truncate=False)


if __name__ == "__main__":
    run()

File: r/sparklabs/batch/sales/dataframe/distinct_customer.R
source("r/sparklabs/batch/sales/dataframe/utils.R")

distinct_customer_count <- count(select(orders, "customer_id"))
expensive_product <- limit(orderBy(products, desc("price")), 1)

cat(sprintf("Total distinct customers: %s\n", distinct_customer_count))
cat("Most Expensive Product:\n")
showDF(expensive_product, truncate = FALSE)
